scripts/backup.sh

```
@@ -0,0 +1,244 @@
+# Health check function
+health_check() {
+    local db_path="$DATA_PATH/$DB_FILE"
+    
+    # Check if database is accessible
+    if ! sqlite3 "$db_path" "SELECT 1;" >/dev/null 2>&1; then
+        log ERROR "Database health check failed: cannot access database"
+        return 1
+    fi
+    
+    # Check disk space
+    local available_space=$(df "$BACKUP_PATH" | tail -1 | awk '{print $4}')
```

Shellcheck (0.10.0)

[warning] Declare and assign separately to avoid masking return values.

(SC2155)